homeassistant/components/sensor/magicseaweed.py

```diff
@@ -0,0 +1,258 @@
+
+DEFAULT_UNIT = 'us'
+DEFAULT_NAME = 'MSW'
+
+API_URL = 'http://magicseaweed.com/api/{}/forecast/'
+
+HOURS = ['12AM', '3AM', '6AM', '9AM', '12PM', '3PM', '6PM', '9PM']
+
+SENSOR_TYPES = {
+    'max_breaking_swell': ['Max'],
+    'min_breaking_swell': ['Min'],
+    'swell_forecast': ['Forecast'],
+}
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_SPOT_ID): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_HOURS, default=None):
+        vol.All(cv.ensure_list, [vol.In(HOURS)]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_UNITS): vol.In(['eu', 'uk', 'us']),
+    vol.Optional(CONF_UPDATE_INTERVAL, default=timedelta(seconds=300)): (
+        vol.All(cv.time_period, cv.positive_timedelta)),
+})
+
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Magicseaweed sensor."""
+    name = config.get(CONF_NAME)
+    spot_id = config.get(CONF_SPOT_ID)
+    api_key = config.get(CONF_API_KEY)
+    hours = config.get(CONF_HOURS)
+
+    if CONF_UNITS in config:
+        units = config.get(CONF_UNITS)
+    elif hass.config.units.is_metric:
+        units = 'eu'
+    else:
+        units = 'us'
+
+    forecast_data = MagicSeaweedData(
+        api_key=api_key,
+        spot_id=spot_id,
+        units=units,
+        interval=config.get(CONF_UPDATE_INTERVAL))
+    forecast_data.update()
+
+    # If connection failed don't setup platform.
+    if forecast_data.currently is None or forecast_data.hourly is None:
+        return False
+
+    sensors = []
+    for variable in config[CONF_MONITORED_CONDITIONS]:
+        if 'forecast' in variable:
+            sensors.append(MagicSeaweedSensor(forecast_data, variable, name,
+                                              units))
+
+        else:
+            sensors.append(MagicSeaweedSensor(forecast_data, variable, name,
+                                              units))
+            if hours is not None:
+                for hour in hours:
+                    sensors.append(MagicSeaweedSensor(
+                        forecast_data, variable, name, units, hour))
+    add_devices(sensors, True)
+
+
+class MagicSeaweedSensor(Entity):
+    """Implementation of a MagicSeaweed sensor."""
+
+    def __init__(self, forecast_data, sensor_type, name, unit_system,
+                 hour=None):
+        """Initialize the sensor."""
+        self.client_name = name
+        self._name = SENSOR_TYPES[sensor_type][0]
+        self.data = forecast_data
+        self.type = sensor_type
+        self.hour = hour
+        self._unit_system = unit_system
+        self._state = None
+        self._icon = None
+        self._unit_of_measurement = None
```

Comment:
It's a small thing, but unless you have a reason for doing so, keep your private properties together and your public properties together.